zcash_primitives/src/note_encryption.rs

```diff
@@ -305,8 +307,10 @@ impl SaplingNoteEncryption {
         self.note
             .r
             .into_repr()
-            .write_le(&mut input[20..COMPACT_NOTE_SIZE])
+            .write_le(&mut input[20..COMPACT_NOTE_SIZE - 4])
```

Comment:
Hmm this seems odd, should we change `COMPACT_NOTE_SIZE` instead perhaps?

Reply:
yes, this seems like a quick change to make it work. more formalization needed.